### PR TITLE
Add button alignment option to product grid blocks

### LINKED ISSUE #384: Button alignment option in product grid blocks
Due to different content lengths the "add to cart" buttons in these blocks can have awkward alignment. Example:

alignment

Let's add an option to force the buttons to align vertically with each other.

### PR DESCRIPTION
**Pull request by mikejolley**
This PR adds a new toggle option to align product add to cart buttons vertically. I've implemented this across all grid based blocks. Tests are passing locally.

Closes #384

#### Accessibility

I tested toggling on/off the option via keyboard controls. I've followed the same pattern as other toggles found in the visibility section.

- [x] I've tested using only a keyboard (no mouse)

### Screenshots

This is how it looks toggled on:
Edit Post ‹ local wordpress test — WordPress 2019-05-28 17-19-55

And toggled off (default):
Edit Post ‹ local wordpress test — WordPress 2019-05-28 17-20-26

When enabled, product cart buttons align like this using flexbox:
Edit Post ‹ local wordpress test — WordPress 2019-05-28 17-20-52

### How to test the changes in this Pull Request:

1. For each of the grid based block types, ensure you can see this option under "layout", toggle the option, and that the option stays toggled on after update and refresh.
2. Check the flexbox works by setting up products with varying title lengths. Buttons should be aligned like the screenshot above.
3. Check (2) on the frontend too - same styling should apply there.

### CODE REVIEW

File: assets/js/blocks/products-by-attribute/index.js [hunk 1]
```diff
@@ -129,6 +137,10 @@ registerBlockType( blockTypeName, {
 					type: 'number',
 					default: wc_product_block_data.default_rows,
 				},
+				alignButtons: {
```
Comment (ryelle) [inline assets/js/blocks/products-by-attribute/index.js]: This shouldn't be necessary, the attributes here are a snapshot of what the attributes were when this was deprecated.

File: assets/php/class-wgpb-block-grid-base.php [hunk 2]
```diff
@@ -227,6 +228,10 @@ protected function get_container_classes() {
 			$classes[] = "align{$this->attributes['align']}";
 		}
 
+		if ( ! empty( $this->attributes['alignButtons'] ) ) {
+			$classes[] = 'align-cart-buttons';
```
Comment (ryelle) [inline assets/php/class-wgpb-block-grid-base.php]: Let's keep the `is-/has-*` standard (`align*` comes from wp core), so maybe `has-aligned-buttons`?

File: assets/js/components/grid-layout-control/index.js [hunk 3]
```diff
@@ -54,6 +70,10 @@ GridLayoutControl.propTypes = {
 	 * The current rows count.
 	 */
 	rows: PropTypes.oneOfType( [ PropTypes.number, PropTypes.string ] ).isRequired,
+	/**
+	 * The current rows count.
```
Comment (ryelle) [inline assets/js/components/grid-layout-control/index.js]: This should be updated.

**Comment by mikejolley** [comment]
@ryelle Updated to master and fixed those issues. Not sure if we need design review too.

**Comment by jwold** [comment]
This design pattern is something we do in a number of other places, so it looks good. Thanks!

**Pull request merged**